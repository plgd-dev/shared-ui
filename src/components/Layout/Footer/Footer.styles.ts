import { css } from '@emotion/react'
import { colors } from '../../Atomic/_utils/colors'
import { fontPrimary } from '../../Atomic/_utils/commonStyles'

export const footer = css`
    border-top: 1px solid #e6e9ed;
    background: #fff;
    padding: 0 40px;
    box-sizing: border-box;
    flex-shrink: 0;
    flex-grow: 0;
    display: flex;
    flex-direction: column;
    overflow: hidden;
    position: relative;
    z-index: 2;
`

export const footerInner = css`
    flex: 0 0 420px;
`

export const footerMainLine = css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px 0;
    min-height: 64px;
`

export const recentTasks = css`
    font-family: ${fontPrimary};
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    text-align: center;
    color: ${colors.neutral900};
    text-decoration: none;
    display: flex;
    align-items: center;
    padding-right: 22px;
    position: relative;
`

export const icon = css`
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    right: 0;
    color: ${colors.neutral500};
    transition: all 0.3s;
`

export const panelOpen = css`
    transform: translateY(-50%) rotate(180deg);
`
