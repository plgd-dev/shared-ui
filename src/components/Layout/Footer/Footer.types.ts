import { ReactNode } from 'react'

export type FooterSizeType = 'small' | 'medium'

export type Props = {
    className?: string
    contentLeft?: ReactNode
    footerExpanded?: boolean
    id?: string
    innerPortalTarget?: ReactNode
    paginationComponent: ReactNode
    recentTasksPortal?: ReactNode
    recentTasksPortalTitle?: ReactNode
    setFooterExpanded?: (expanded: boolean) => void
    size?: FooterSizeType
}

export const defaultProps: Partial<Props> = {
    size: 'small',
}
