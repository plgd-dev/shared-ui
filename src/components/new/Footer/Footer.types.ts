export type FooterLinkType = {
    to: string
    i18key: string
}

export type Props = {
    links: FooterLinkType[]
}
