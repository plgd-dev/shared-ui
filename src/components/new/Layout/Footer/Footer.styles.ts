import { css } from '@emotion/react'

export const footer = css`
    height: 65px;
    border-top: 1px solid #e6e9ed;
    background: #fff;
    padding: 16px 40px;
    box-sizing: border-box;
    display: flex;
    align-items: center;
    justify-content: space-between;
`
