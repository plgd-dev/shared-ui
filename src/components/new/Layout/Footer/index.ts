export { default } from './Footer'
export * from './Footer'

export { default as Pagination } from './Pagination'
export * from './Pagination'
