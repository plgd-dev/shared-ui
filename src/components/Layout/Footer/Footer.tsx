import { FC, memo } from 'react'
import { motion } from 'framer-motion'
import isFunction from 'lodash/isFunction'
import { Props } from './Footer.types'
import * as styles from './Footer.styles'
import { Icon } from '../../Atomic/Icon'

const Footer: FC<Props> = memo((props) => {
    const { paginationComponent, recentTasksPortal, recentTasksPortalTitle, footerExpanded, setFooterExpanded } = props

    const handleToggle = (e: any) => {
        e.preventDefault()
        isFunction(setFooterExpanded) && setFooterExpanded(!footerExpanded)
    }

    const LeftSide = () => {
        if (recentTasksPortalTitle) {
            return (
                <a css={styles.recentTasks} href='packages/shared-ui/src/components/Layout/Footer#' onClick={handleToggle}>
                    {recentTasksPortalTitle} <Icon css={[styles.icon, footerExpanded && styles.panelOpen]} icon='arrow-triangle-full-up' />
                </a>
            )
        } else {
            return <div />
        }
    }

    return (
        <motion.div
            layout
            animate={{
                height: footerExpanded ? 420 : 65,
                flex: `0 0 ${footerExpanded ? 420 : 65}`,
            }}
            css={styles.footer}
            exit={{
                height: 65,
            }}
            initial={false}
            transition={{
                duration: 0.3,
            }}
        >
            <div css={styles.footerInner}>
                <div css={styles.footerMainLine}>
                    <LeftSide />
                    <div>{paginationComponent}</div>
                </div>

                {recentTasksPortal}
            </div>
        </motion.div>
    )
})

Footer.displayName = 'Footer'

export default Footer
