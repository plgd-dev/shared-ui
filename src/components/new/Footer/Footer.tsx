import { FC, memo, useState } from 'react'
import { useIntl } from 'react-intl'
import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
import Tooltip from 'react-bootstrap/Tooltip'
import Popover from 'react-bootstrap/Popover'
import { format } from 'date-fns'
import { messages as t } from './Footer.i18n'
import './Footer.scss'
import { useAppConfig } from '@/containers/App'
import { copyToClipboard } from '../../../common/utils'
import { showSuccessToast } from '../../old/toast'
import Display from '../Display'
import { FooterLinkType, Props } from './Footer.types'

const Footer: FC<Props> = memo(({ links }) => {
    const { formatMessage: _ } = useIntl()
    const { buildInformation } = useAppConfig()
    const [show, setShow] = useState(false)

    const handleOnMouseEnter = () => {
        setShow(true)
    }
    const handleOnMouseLeave = () => {
        setShow(false)
    }

    const copyVersion = () => {
        if (buildInformation?.version) {
            copyToClipboard(buildInformation.version)
            showSuccessToast({
                title: _(t.done),
                message: _(t.copied),
            })
        }
    }

    const BuildInformation = () => {
        if (!buildInformation?.buildDate) {
            return null
        }
        return (
            <Display when={!!buildInformation}>
                <OverlayTrigger
                    placement='top'
                    overlay={
                        <Tooltip className='plgd-tooltip' id='footer-version'>
                            {_(t.clickToCopy)}
                        </Tooltip>
                    }
                >
                    <span className='copy' onClick={copyVersion}>
                        <i className={`fas fa-copy m-r-10`} />
                    </span>
                </OverlayTrigger>

                <OverlayTrigger
                    placement='top-start'
                    show={show}
                    delay={{ show: 0, hide: 1000 }}
                    overlay={
                        <Popover
                            id='popover-positioned-top'
                            className='plgd-popover interactive'
                            onMouseEnter={handleOnMouseEnter}
                            onMouseLeave={handleOnMouseLeave}
                        >
                            <Popover.Body>
                                <div className='footerBuildInfo'>
                                    <div className='line'>
                                        <div className='attr'>Build time:</div>
                                        <div className='val'>{format(new Date(buildInformation.buildDate), 'Pp')}</div>
                                    </div>
                                    <div className='line'>
                                        <div className='attr'>Commit hash:</div>
                                        <a
                                            className='val'
                                            href={`https://github.com/plgd-dev/client-application/commit/${buildInformation.commitHash}`}
                                            target='_blank'
                                        >
                                            {buildInformation.commitHash}
                                        </a>
                                    </div>
                                </div>
                            </Popover.Body>
                        </Popover>
                    }
                >
                    <span onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave}>
                        {buildInformation.version}
                    </span>
                </OverlayTrigger>
            </Display>
        )
    }

    return (
        <footer id='footer'>
            <div id='pending-commands-wrapper'></div>
            <div className='left'>
                <BuildInformation />
            </div>
            <div className='right'>
                {links.map((item: FooterLinkType, index) => {
                    // @ts-ignore
                    const tran = _(t[item.i18key])
                    return (
                        <a href={item.to} target='_blank' rel='noopener' key={index}>
                            {tran}
                        </a>
                    )
                })}
            </div>
        </footer>
    )
})

Footer.displayName = 'Footer'

export default Footer
