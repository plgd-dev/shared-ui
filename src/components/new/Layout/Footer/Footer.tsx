import { FC, memo } from 'react'
import { Props } from './Footer.types'
import * as styles from './Footer.styles'
import { Icon } from '../../Icon'
import { motion } from 'framer-motion'

const Footer: FC<Props> = memo((props) => {
    const { paginationComponent, recentTasksPortal, recentTasksPortalTitle, footerExpanded, setFooterExpanded } = props

    const handleToggle = (e: any) => {
        e.preventDefault()
        setFooterExpanded(!footerExpanded)
    }

    const LeftSide = () => {
        if (recentTasksPortalTitle) {
            return (
                <a css={styles.recentTasks} href='#' onClick={handleToggle}>
                    {recentTasksPortalTitle} <Icon css={[styles.icon, footerExpanded && styles.panelOpen]} icon='arrow-triangle-full-up' />
                </a>
            )
        } else {
            return <div />
        }
    }

    return (
        <motion.div
            layout
            animate={{
                height: footerExpanded ? 420 : 65,
                flex: `0 0 ${footerExpanded ? 420 : 65}`,
            }}
            css={styles.footer}
            exit={{
                height: 65,
            }}
            initial={false}
            transition={{
                duration: 0.3,
            }}
        >
            <div css={styles.footerMainLine}>
                <LeftSide />
                <div>{paginationComponent}</div>
            </div>

            {recentTasksPortal}
        </motion.div>
    )
})

Footer.displayName = 'Footer'

export default Footer
