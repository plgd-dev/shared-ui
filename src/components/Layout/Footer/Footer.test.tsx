import React from 'react'
import { render } from '../../../../test/jest-wrapper'
import Footer from './Footer'

describe('<Footer>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <Footer
                paginationComponent={<div id='paginationPortalTarget'></div>}
                recentTasksPortal={<div id='recentTasksPortalTarget'></div>}
                recentTasksPortalTitle={<span id='recentTasksPortalTitleTarget'>Recent tasks</span>}
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
