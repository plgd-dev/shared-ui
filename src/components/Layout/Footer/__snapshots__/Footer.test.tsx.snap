// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-top: 1px solid #e6e9ed;
  background: #fff;
  padding: 0 40px;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
}

.emotion-1 {
  -webkit-flex: 0 0 420px;
  -ms-flex: 0 0 420px;
  flex: 0 0 420px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 16px 0;
  min-height: 64px;
}

.emotion-3 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 18px;
  text-align: center;
  color: #191A1A;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 22px;
  position: relative;
}

.emotion-4 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 0;
  color: #81868C;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

<div
    class="emotion-0"
    style="height: 65px;"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          href="packages/shared-ui/src/components/Layout/Footer#"
        >
          <span
            id="recentTasksPortalTitleTarget"
          >
            Recent tasks
          </span>
           
          <svg
            class="emotion-4"
            fill="none"
            height="16"
            viewBox="0 0 6 11"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M.126 6.854a.387.387 0 0 0 .606 0L3 4.207l2.268 2.647a.387.387 0 0 0 .606 0 .56.56 0 0 0 0-.708l-2.571-3a.387.387 0 0 0-.606 0l-2.571 3a.56.56 0 0 0 0 .708Z"
              fill="currentcolor"
              fill-rule="evenodd"
            />
            <g
              clip-path="url(#icon-arrow-triangle-full-up_svg__a)"
            >
              <path
                d="M2.576 3.196a.56.56 0 0 1 .85 0l2.4 2.666a.718.718 0 0 1 .129.727c-.094.25-.311.413-.555.413L.6 7c-.242 0-.461-.163-.555-.413a.723.723 0 0 1 .13-.727l2.4-2.666.001.002Z"
                fill="currentcolor"
              />
            </g>
            <defs>
              <clippath
                id="icon-arrow-triangle-full-up_svg__a"
              >
                <path
                  d="M0 0h6v10.667H0z"
                  fill="#fff"
                  transform="matrix(1 0 0 -1 0 11)"
                />
              </clippath>
            </defs>
          </svg>
        </a>
        <div>
          <div
            id="paginationPortalTarget"
          />
        </div>
      </div>
      <div
        id="recentTasksPortalTarget"
      />
    </div>
  </div>
</DocumentFragment>
`;
